tox.ini

```diff
@@ -1,37 +1,46 @@
 [tox]
+skipsdist = True
 envlist =
     docs, packaging, lint-py2, lint-py3, mypy,
     py27, py34, py35, py36, py37, py38, pypy, pypy3
 
+[shorthands]
+pip = python {toxinidir}/tools/tox_pip.py
+
 [testenv]
+skip_install = True
+# Environment variables
 passenv = CI GIT_SSL_CAINFO
 setenv =
-    # This is required in order to get UTF-8 output inside of the subprocesses
-    # that our tests use.
+    # Required to get UTF-8 output inside of the subprocesses in tests.
     LC_CTYPE = en_US.UTF-8
+# Use tox_pip.py for when tox runs pip.
+install_command = {[shorthands]pip} install {opts} {packages}
+list_dependencies_command = {[shorthands]pip} freeze
+# The regular stuff
 deps = -r{toxinidir}/tools/tests-requirements.txt
-commands = pytest --timeout 300 []
-install_command = python -m pip install {opts} {packages}
-usedevelop = True
+commands =
+    {[shorthands]pip} install -e {toxinidir}
```

Comment:
tox runs `freeze` _after_ installing develop if `usedevelop = True`.

This is why this is a separate line in commands.

Reply:
We _could_ be "smart" in the wrapper script and not overwrite if we're doing a freeze but this just feels cleaner (as we can then just remove multiple `skip_install = True` lines and add the above editable install line to simply install the development version, when needed.